black: add page (#3506)
tldr-pages/tldr: pages/common/black.md

# black

> > A speedy Python auto code formatter.

Reviewer: I would say we remove the word "speedy" as we try to omit any adjectives from the command description. Writing with adjectives almost feels like an ad.
```suggestion
> A Python auto code formatter.
```

> > More information <https://github.com/psf/black>.

Reviewer 2: Missing a colon after information.

Author: resolved

- Auto-format a file or entire directory:

> `black {source_file_or_directory}`

Reviewer 3: Looks like this should be using double curly braces to match the style guide, could you also update the references to use `path/to/file_or_directory` throughout the page? 🙂

```suggestion
`black {{path/to/file_or_directory}}`
```

- Format the code passed in as a string:

> `black -c {{source_file_or_directory}}`

Reviewer 3: ```suggestion
`black -c {{path/to/file_or_directory}}`
```

- Output a diff for each file on stdout:

> `black --diff {{source_file_or_directory}}`

Reviewer 3: ```suggestion
`black --diff {{path/to/file_or_directory}}`
```

> - Don't write the files back, just return the status:

Reviewer: I feel like we should always start or descriptions with a verb.
```suggestion
- Return the status without writing the files back:
```
Not sure if that's the best way to rephrase it, though.

> `black --check {{source_file_or_directory}}`

Reviewer 3: ```suggestion
`black --check {{path/to/file_or_directory}}`
```

> - Don't emit non-error messages to stderr:

Reviewer: Once again the verb thing: 
```suggestion
- Auto-format a file or directory without emitting non-error messages to stderr:
```

Reviewer 4: That double negation feels awkward, I'd probably suggest:

    - Auto-format a file or directory emitting only error messages to stderr:

> - Auto-format a file or directory without emitting non-error messages to stderr:

Reviewer: Actually I got this extra one: 
Instead of using a double negative, let's write it this way. 
```suggestion
- Auto-format a file or directory emitting exclusively error messages to stderr:
```

> `black --quiet {{source_file_or_directory}}`

Reviewer 3: ```suggestion
`black --quiet {{path/to/file_or_directory}}`
```
